### PR TITLE
Remove guidance on migrating to http endpoints and data api

### PR DESCRIPTION
**Pull request by MongoCaleb**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-41041

- [Use a VPC Private Endpoint (Preview)](https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/DOCSP-41041/security/private-endpoints/#migrate-a-legacy-endpoint): Remove guidance on migrating to http endpoints and data api
- [Custom HTTPS Endpoints](https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/DOCSP-41041/data-api/custom-endpoints/#std-label-custom-endpoints): add note.
- [Data API Endpoints](https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/DOCSP-41041/data-api/generated-endpoints/#std-label-data-api-endpoints): add note.

### Release Notes
- **Private Endpoints**
  - Private endpoints do not support HTTPS Endpoints or the Data API.

### CODE REVIEW

File: source/security/private-endpoints.txt [hunk 1]
```diff
@@ -67,6 +71,11 @@ You can add a new Private Endpoint with the App Services UI or the Admin API.
 You can add a new General Endpoint or a new Sync Endpoint. You can also
 migrate a Legacy Endpoint to either a new General or Sync Endpoint.
 
+.. note:: 
+   
+   Endpoints that use the ``realm.mongodb.com`` domain must migrate to use the new
+   ``services.cloud.mongodb.com`` domain.
+
```
Comment (melodeeli98) [inline source/security/private-endpoints.txt:78]: I think we should remove all references to the legacy endpoints and their migrations (endpoints that use the realm domain are legacy endpoints)

**Review by melodeeli98 — APPROVED** [review]
lgtm pending one comment, thank you for the quick turnaround!

**Review by krollins-mdb — APPROVED** [review]
LGTM. Thanks for the quick turnaround, @MongoCaleb!

**Comment by krollins-mdb** [comment]
Oh, @MongoCaleb, can you add a release notes section? Like what we have in Dachary's PR here: https://github.com/mongodb/docs-app-services/pull/756?

It makes putting together the docs release notes much easier.

**Pull request merged**